### PR TITLE
Error Handling for pvc creation from copysnapshot

### PR DESCRIPTION
**Pull request by HarishH-DELL**
# PR Submission checklist
 Error Handing for PVC Creation from Copysnapshot
# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/749|

# Common PR Checklist:

- [X] Have you made sure that the code compiles?
- [X] Have you commented your code, particularly in hard-to-understand areas
- [X] Did you run tests in a real Kubernetes cluster?
- [X] Have you maintained backward compatibility

## Description of your changes:
Taken the Response from the OneFS, checking the weather response as success state as false, which means copy snapshot is failed. and collecting error messages from response JSON 

In Failed case
image
image
Inked230594379-08fd99f9-da60-4278-92e7-518b01269d24

image

In Success Case
image
image

### CODE REVIEW

**Review by chimanjain — APPROVED** [review]

File: snapshots.go [hunk 1]
```diff
@@ -151,9 +154,24 @@ func (c *Client) CopySnapshotWithIsiPath(
...
 
-	_, err = api.CopyIsiSnapshotWithIsiPath(
+	resp, err := api.CopyIsiSnapshotWithIsiPath(
 		ctx, c.API, isiPath, snapshotSourceVolumeIsiPath, snapshot.Name,
 		path.Base(snapshot.Path), destinationName)
+
+	//The response will be null on success of the snapshot creation otherwise it will return the response with a success state equal to false and with details
+	if resp != nil && !resp.Success {
+		var copySnapError bytes.Buffer
+		if resp.Errors != nil {
```
Comment (rajkumar-palani) [inline snapshots.go]: move this if condition to upper if.

File: snapshots.go [hunk 2]
```diff
@@ -151,9 +154,24 @@ func (c *Client) CopySnapshotWithIsiPath(
...
+			for _, errMes := range resp.Errors {
+				//Extracting the  error message from the JSON array
+				copySnapError.WriteString("Error Source = " + errMes.Source + "," + "Message = " + errMes.Message + "," + "," + "Source = " + errMes.Source + "," + "Target = " + errMes.Target + " \n")
+			}
+		}
+
+		err = errors.New(copySnapError.String())
```
Comment (rajkumar-palani) [inline snapshots.go:169]: declare the copySnapError variable inside if and assign the error.

**Review by rajkumar-palani — APPROVED** [review]
lgtm

**Review by shefali-malhotra — DISMISSED** [review]
Please remove all the Ip's from result screenshots . 
Please do run round of unit test with your changes.

File: snapshots.go (hunk 1 above)
Comment (HarishH-DELL) [reply snapshots.go]: moved

File: snapshots.go (hunk 2 above)
Comment (HarishH-DELL) [reply snapshots.go:169]: done

**Comment by HarishH-DELL** [comment]
> Please remove all the Ip's from result screenshots . Please do run round of unit test with your changes.

image
Removed the  Ip's from result screenshots and run the make unit-test and it success

**Comment by nitesh3108** [comment]
> Please remove all the Ip's from result screenshots . Please do run round of unit test with your changes.

addressed

**Review by shefali-malhotra — APPROVED** [review]
LGTM

**Pull request merged**